cmd/nvidia-cdi-hook/cudacompat/cudacompat_test.go

```diff
@@ -0,0 +1,131 @@
+/*
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+*/
+
+package cudacompat
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	testlog "github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCompatLibs(t *testing.T) {
+	logger, _ := testlog.NewNullLogger()
+
+	testCases := []struct {
+		description            string
+		contents               map[string]string
+		options                options
+		expectedCompatContents string
+	}{
+		{
+			description: "empty root",
+			options: options{
+				hostDriverVersion: "222.55.66",
+			},
+		},
+		{
+			description: "compat lib is newer; no ldcache",
+			contents: map[string]string{
+				"/usr/local/cuda/compat/libcuda.so.333.88.99": "",
+			},
+			options: options{
+				hostDriverVersion: "222.55.66",
+			},
+		},
+		{
+			description: "compat lib is newer; ldcache",
+			contents: map[string]string{
+				"/etc/ld.so.cache":                            "",
+				"/etc/ld.so.conf.d/.hidden":                   "",
+				"/usr/local/cuda/compat/libcuda.so.333.88.99": "",
+			},
+			options: options{
+				hostDriverVersion: "222.55.66",
+			},
+			expectedCompatContents: "/usr/local/cuda/compat\n",
+		},
+		{
+			description: "compat lib is older; ldcache",
+			contents: map[string]string{
+				"/etc/ld.so.cache":                            "",
+				"/etc/ld.so.conf.d/.hidden":                   "",
+				"/usr/local/cuda/compat/libcuda.so.111.88.99": "",
+			},
+			options: options{
+				hostDriverVersion: "222.55.66",
+			},
+			expectedCompatContents: "",
+		},
+		{
+			description: "compat lib has same major version; ldcache",
+			contents: map[string]string{
+				"/etc/ld.so.cache":                            "",
+				"/etc/ld.so.conf.d/.hidden":                   "",
+				"/usr/local/cuda/compat/libcuda.so.222.88.99": "",
+			},
+			options: options{
+				hostDriverVersion: "222.55.66",
+			},
+			expectedCompatContents: "",
+		},
+		{
+			description: "driver version empty; ldcache",
```

Comment:
Question -- I am just trying to grok this scenario. IIUC this means we always use the compat libraries if we cannot discover the driver version of the libraries present on the host. Does libnvidia-container have the same behavior? Taking a step back, shouldn't failing to detect the driver version from the host not result in an error?

Reply:
Yes, failing to detect the driver version will cause a failure in `libnvidia-container`. That behaviour is not captured here since the driver version extraction happens earlier in the process.

The reason that I didn't want to raise an error in this case is that I didn't want to cause a container to fail to start due to an error in the runtime -- more drastically changing the behaviour. I think I can change the behaviour of the hook to be a no-op if the driver version is not specified though.

Reply:
I have change the behaviour to a no-op if the driver root is empty.